client/src/components/profile-box.js

```diff
@@ -14,7 +14,7 @@ const ProfileBox = ({
     if (!user) return;
     Object.keys(getFieldsValue()).forEach(key => {
       const obj = {};
-      obj[key] = user[key] || null;
+      obj[key] = user[key] || '';
```

Comment:
Instead of setting the default value to an empty string, we should exclude empty profile field from the payload.

Check out https://stackoverflow.com/questions/30812765/how-to-remove-undefined-and-null-values-from-an-object-using-lodash 

You likely need to install a new package `lodash.identity`.